himan-scripts/cloud-consensus.lua

```diff
@@ -0,0 +1,223 @@
+  local stDev = math.sqrt(math.max(((wt_ens_ec * (N_STD_EC^2 + N_MEAN_EC^2) + wt_ec * N_EC^2) / wt_sum) - mean^2, 0))
+  return mean, stDev
+end
+
+function compute_std_mean_all(N_EC, N_MEPS, N_MEAN_EC, N_MEAN_MEPS, N_STD_EC, N_STD_MEPS, wt_ec, wt_ens_ec, wt_meps, wt_ens_meps, wt_sum)
+  local mean = (N_EC * wt_ec + N_MEAN_EC * wt_ens_ec + N_MEPS * wt_meps + N_MEAN_MEPS * wt_ens_meps) / wt_sum
+  local stDev = math.sqrt(math.max(((wt_ens_ec * (N_STD_EC^2 + N_MEAN_EC^2) + wt_ec * N_EC^2 + wt_meps * N_MEPS^2 + wt_ens_meps * (N_STD_MEPS^2 + N_MEAN_MEPS^2)) / wt_sum) - mean^2, 0))
+  return mean, stDev
+end
+
+function convert_to_100(array)
+  if not array then return nil end 
+  local t = {}
+  for i = 1, #array do
+      t[i] = array[i] * 100
+  end
+  return t
+end
+
+function process_params(producer, ftype, ...)
+  local params = { ... }
+  local results = { get_param(producer, ftype, table.unpack(params)) }
+  
+  -- Convert all results to 100-scale
+  for i = 1, #results do
+      results[i] = convert_to_100(results[i])
+  end
+  
+  return table.unpack(results)  
+end
+
+function get_data(producer1, producer2, ftype)
+  local mean_params = {
+      param("NL-MEAN-0TO1", aggregation(), processing_type(HPProcessingType.kMean)),
+      param("NM-MEAN-0TO1", aggregation(), processing_type(HPProcessingType.kMean)),
+      param("NH-MEAN-0TO1", aggregation(), processing_type(HPProcessingType.kMean))
+  }
+
+  local std_params = {
+      param("NL-STDDEV-0TO1"),
+      param("NM-STDDEV-0TO1"),
+      param("NH-STDDEV-0TO1")
+  }
+
+  local main_params = {
+      param("NL-0TO1"),
+      param("NM-0TO1"),
+      param("NH-0TO1")
+  }
+
+  local NL_MEAN, NM_MEAN, NH_MEAN = process_params(producer2, forecast_type(HPForecastType.kStatisticalProcessing), table.unpack(mean_params))
+  local NL_STD, NM_STD, NH_STD = process_params(producer2, forecast_type(HPForecastType.kStatisticalProcessing), table.unpack(std_params))
+  local NL, NM, NH = process_params(producer1, ftype, table.unpack(main_params))
+
+  return NL, NM, NH, NL_MEAN, NM_MEAN, NH_MEAN, NL_STD, NM_STD, NH_STD
+end
+
+-- Get origin times for MEPS and EC
+function get_time(producer)
+  
+  local test = configuration:Exists("origin_time_test")
+
+  if test then
+    local test_time = configuration:GetValue("origin_time_test")
+    ftime = forecast_time(raw_time(test_time), raw_time("2025-04-04 06:00:00"))
+    return ftime
+  end
+
+  local vire_hour = current_time:GetOriginDateTime():String('%H')
+  local producer_id = producer:GetId()
+
+  if producer_id == 4 or producer_id == 260 then
+    adjust_hours = -4
+  elseif (vire_hour == '07' or vire_hour == '19') and (producer_id == 131 or producer_id == 242) then
+      adjust_hours = -7
+  elseif (vire_hour == '13' or vire_hour == '01') and (producer_id == 131 or producer_id == 242) then
+      adjust_hours = -13
+  end
+
+  if adjust_hours then
+    current_time:GetOriginDateTime():Adjust(HPTimeResolution.kHourResolution, adjust_hours)
```

Comment:
To me this is a bit fishy... you are changing a value of a global variable, creating a local variable based on that, and the resetting the global. Why not first create the local variable and modify that?

```
local adjust_hours = 0

if producucer_id == 4 or producer_id == 260 then
  ...
end

local ftime = forecast_time(current_time:GetOriginDateTime(), current_time:GetValidDateTime())
ftime:GetOriginDateTime():Adjust(HPTimeResolution.kHourResolution, adjust_hours)

return ftime
```